.github/workflows/ci-repo.yml

```
@@ -39,43 +51,135 @@ jobs:
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
 
-  vscode-tests:
-    runs-on: ubuntu-latest
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: "vscode/.nvmrc"
+
+      - name: Install dependencies
+        working-directory: ./vscode
+        run: npm ci
+
+      # Run tests on Linux
+      - name: Run tests (Linux)
+        if: matrix.arch == 'linux'
+        working-directory: ./vscode
+        run: 
+          xvfb-run -a npm run test
+
+      # Run tests on macOS
+      - name: Run tests (macOS)
+        if: matrix.arch == 'macos'
+        working-directory: ./vscode
+        run: npm test
+
+      # Run tests on Windows
+      - name: Run tests (Windows)
+        if: matrix.arch == 'windows'
+        working-directory: ./vscode
+        shell: cmd
+        run: npm test
+
+  package:
+    name: Package (${{ matrix.arch }})
+    runs-on: ${{ matrix.os }}
+    needs: test
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            arch: linux
+          - os: macos-latest
+            arch: macos
+          - os: windows-latest
+            arch: windows
+      max-parallel: 3
 
     steps:
-      - name: Check-out repository
+      - name: Checkout code
         uses: actions/checkout@v4
 
-      - name: Use Node.js from .nvmrc
+      - name: Set up Node.js
         uses: actions/setup-node@v4
         with:
           node-version-file: "vscode/.nvmrc"
 
-      - name: Cache extension node_modules
-        uses: actions/cache@v3
+      - name: Install dependencies
+        working-directory: ./vscode
+        run: npm ci
+
+      - name: Build Package
+        working-directory: ./vscode
+        run: npm run package
+
+      - name: Generate .vsix package
+        working-directory: ./vscode
+        run: |
+          npm install @vscode/vsce
+          npx vsce package
+
+      - name: Upload VSIX artifact
+        uses: actions/upload-artifact@v4
         with:
-          path: vscode/node_modules
-          key: ${{ runner.os }}-node-${{ hashFiles('vscode/package-lock.json') }}
+          name: vscode-extension-${{ matrix.arch }}
+          path: "vscode/*.vsix"
+
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    needs: package
+    if: github.ref == 'refs/heads/main'
 
-      - run:  |
-            cd vscode
-            npm ci
-      - run: |
-            cd vscode
-            xvfb-run -a npm run test
-      
\ No newline at end of file
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Download VSIX artifacts
+        uses: actions/download-artifact@v3
+        with:
+          path: ./artifacts
+
+      - name: Generate tag name
+        id: vars
+        shell: bash
+        run: |
+          SHORT_SHA=$(echo "${GITHUB_SHA}" | cut -c1-8)
+          TAG_NAME="v0.0.1-dev-${SHORT_SHA}"
+          echo "tag_name=$TAG_NAME" >> $GITHUB_OUTPUT
```

We can do this so we actually generate a legit semver release that moves about.

```suggestion
      - name: Generate tag name with timestamp
        id: vars
        shell: bash
        run: |
          TAG_NAME="v0.0.1-dev+$(date +%Y%m%d)"
          echo "tag_name=$TAG_NAME" >> $GITHUB_OUTPUT
```

---

If you think it should be `nightly` instead of `dev`, that's cool, I just don't think we should make it variable at this stage.

---

I am cool with dev.